pkg/jobserver/handler.go

```
@@ -0,0 +1,122 @@
+package jobserver
+
+import (
+	"encoding/json"
+	"net/http"
+	"teleport-jobworker/pkg/job"
+)
+
+// StartReq defines the Start request body.
+type StartReq struct {
+	Program string   `json:"program"`
+	Args    []string `json:"args"`
+}
+
+// SimpleRes defines the Start and Stop response body.
+type SimpleRes struct {
+	ID    string  `json:"id"`
+	Error *string `json:"error"`
+}
+
+// StatusRes defines the GetStatus response body.
+type StatusRes struct {
+	ID       string  `json:"id"`
+	Status   string  `json:"status"`
+	ExitCode *int    `json:"exitCode"`
+	Error    *string `json:"error"`
+}
+
+// OutputRes defines the GetOutput response body.
+type OutputRes struct {
+	ID     string  `json:"id"`
+	Stdout string  `json:"stdout"`
+	Stderr string  `json:"stderr"`
+	Error  *string `json:"error"`
+}
+
+// ErrRes defines error response body for status codes: 401, 404, 500.
+type ErrRes struct {
+	Error string `json:"error"`
+}
+
+// resJSON prepares the response body as JSON.
+func resJSON(w http.ResponseWriter, payload any, code int) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	json.NewEncoder(w).Encode(payload)
```

What happens if json encoding fails? The error is silently ignored.

---

Since I am using the function internally, the payloads should be properly formed. But I have included an error check to make it transparent.